implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -213,21 +193,80 @@ private void loadContents() throws IOException {
-                if (!line.startsWith("-----BEGIN") && !line.startsWith("-----END")) {
-                    // Skip any pem file header/footer
-                    contents.write(line);
+            String line = null;
+            while ((line = reader.readLine()) != null) {
+                contents.write(line);
+            }
+        }
+        return contents.toString();
+    }
+
+    static PublicKey tryAsPEMPublicKey(String content) {
+        LOGGER.debugf("Trying to create a key from the encoded PEM key...");
+        try {
+            return KeyUtils.decodePublicKey(content);
+        } catch (Exception e) {
+            LOGGER.debug("Failed to create a key from the encoded PEM key", e);
+        }
+        return null;
+    }
+
+    static PublicKey tryAsPEMCertificate(String content) {
+        LOGGER.debugf("Trying to create a key from the encoded PEM certificate...");
+        try {
+            return KeyUtils.decodeCertificate(content);
+        } catch (Exception e) {
+            LOGGER.debug("Failed to to create a key from the encoded PEM certificate", e);
+        }
+        return null;
+    }
+
+    private static List<JsonWebKey> loadLocalJwks(String content) throws Exception {
+        LOGGER.debugf("Trying to load the local JWK(S)...");
+
+        JsonObject jwks = null;
+        try (JsonReader reader = Json.createReader(new StringReader(content))) {
+            jwks = reader.readObject();
+        } catch (Exception ex) {
+            LOGGER.debug("Failed to load the local JWK(S)");
+            return null;
+        }
+
+        List<JsonWebKey> localKeys = null;
+        JsonArray keys = jwks.getJsonArray(JsonWebKeySet.JWK_SET_MEMBER_NAME);
+        if (keys != null) {
+            // JWK set
+            localKeys = new ArrayList<>(keys.size());
+            for (int i = 0; i < keys.size(); i++) {
+                localKeys.add(createJsonWebKey(keys.getJsonObject(i)));
+            }
+        } else {
+            // single JWK
+            localKeys = Collections.singletonList(createJsonWebKey(jwks));
+        }
+        return localKeys;
+    }
+
+    private static PublicKey getKeyFromJsonWebKeys(String kid, List<JsonWebKey> keys) {
+        if (kid != null) {
+            for (JsonWebKey currentJwk : keys) {
+                if (kid.equals(currentJwk.getKeyId())) {
+                    return PublicJsonWebKey.class.cast(currentJwk).getPublicKey();
                 }
-                line = reader.readLine();
             }
+        } else if (keys.size() == 1) {
+            return PublicJsonWebKey.class.cast(keys.get(0)).getPublicKey();
         }
-        content = contents.toString();
+        return null;
+    }
+
+    private static JsonWebKey createJsonWebKey(JsonObject jsonObject) throws Exception {
+        return JsonWebKey.Factory.newJwk(JsonUtil.parseJson(jsonObject.toString()));
     }
 
     private static InputStream getAsFileSystemResource(String publicKeyLocation) throws IOException {
```

Comment:
Body of this method should be:

```java
        try {
            return new FileInputStream(publicKeyLocation);
        } catch (FileNotFoundException e) {
            return null;
        }
```

This `return f.exists() ? new FileInputStream(f) : null;` just creates a false sense of security, but doesn't really prevent throwing.

Reply:
Or at the very least, wrap the `return f.exists() ? new FileInputStream(f) : null;` into a `try` block, if you want to minimize number of thrown exceptions.